plugin/switch.vim

```diff
@@ -227,6 +227,23 @@ function! s:SwitchReverse()
   silent! call repeat#set(":Switch\<cr>")
 endfunction
 
+command! -nargs=* SwitchExtend call s:SwitchExtend(<args>)
+fun! s:SwitchExtend(...) abort
+  let b:switch_custom_definitions = get(b:, 'switch_custom_definitions',
+      \                                 copy(get(g:, 'switch_custom_definitions', [])))
+  if a:0 == 0
+    echo b:switch_custom_definitions
+  else
+    for def in a:000
+      if (type(def) == type({}) || type(def) == type([])) &&
+            \ index(b:switch_custom_definitions, def) < 0
+        call extend(b:switch_custom_definitions, [def])
+      endif
```

Comment:
I would like to put an else-clause here to ensure that if the command is called with something unexpected, it tells that to the user. Maybe something like `echoerr "Unexpected definition type, should be a list or a dict: ".string(def)`. What do you think? It could be useful if you make a typo or something.

Reply:
In the first commit I was echoing a message for wrong types, I dropped it because now multiple definitions can be added.
If put in the for loop, I would use `echomsg` rather than `echoerr` because it's an `abort` function and it would terminate, in the end, if the type is wrong nothing happens, it just isn't added, so I would use `echomsg`, or drop the `abort` in the function definition.

Reply:
I'm using `echomsg` with `ErrorMsg` highlight, and showing a message also for duplicate definitions. It's ok?

Reply:
Okay, let's go with this then :+1: